presto-hive/src/main/java/com/facebook/presto/hive/CachingDirectoryLister.java

```diff
@@ -45,23 +44,19 @@
     private final Cache<Path, List<HiveFileInfo>> cache;
     private final Set<SchemaTableName> cachedTableNames;
 
-    protected final DirectoryLister delegate;
-
     @Inject
-    public CachingDirectoryLister(@ForCachingDirectoryLister DirectoryLister delegate, HiveClientConfig hiveClientConfig)
```

Comment:
We may still need the delegate model

Reply:
In future yes, but not required for now.